metadata-ingestion/src/datahub/ingestion/source/metabase.py

```diff
@@ -315,6 +315,46 @@ def emit_card_mces(self) -> Iterable[MetadataWorkUnit]:
             )
             return None
 
+    def emit_card_id(self, card_id: int) -> Source:
+        card_url = f"{self.config.connect_uri}/api/card/{card_id}"
+        try:
+            card_response = self.session.get(card_url)
+            card_response.raise_for_status()
+            card_details = card_response.json()
+
+            return card_details
+        except HTTPError as http_error:
+            self.report.report_warning(
+                key=f"metabase-card-{card_id}",
+                reason=f"Unable to retrieve Card info. " f"Reason: {str(http_error)}",
+            )
+            return None
+
+    def construct_source_path(self, source_paths: set, card_details: dict):
+        try:
+            raw_query = (
+                card_details.get("dataset_query", {}).get("native", {}).get("query", "")
+            )
+            parser = LineageRunner(raw_query)
+
+            for table in parser.source_tables:
+                sources = str(table).split(".")
+                source_schema, source_table = sources[-2], sources[-1]
```

Comment:
Is this always safe? Will there always be schema, table?

Reply:
yes, always a schema return from here ->`source_table_id.startswith("card__")`